### ISSUE TITLE
Improve URL validation

### ISSUE THREAD

**Issue by ernilambar** [10up/plugin-check#890]
We have pretty basic URL validation currently:
```php
function is_valid_url( $url ) {
	return filter_var( $url, FILTER_VALIDATE_URL ) === $url && str_starts_with( $url, 'http' );
}
```

We frequently get this type of URL with duplicated protocol but our function does not detect that.
```
https://http://example.com/
```

We could improve the `is_valid_url()` function to handle such edge case.
